core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequests.scala

```diff
@@ -38,6 +39,8 @@ private[spark] class ExecutorResourceRequests() extends Serializable {
 
   def requests: Map[String, ExecutorResourceRequest] = _executorResources.asScala.toMap
 
+  def requestsJMap: JMap[String, ExecutorResourceRequest] = requests.asJava
```

Comment:
If you target to support Java API from this class, it might be just best to let `requests` return a Java map directly. In Scala side, it's pretty easily able to use via using `asScala`. I think this is the usual approach in the case base if I am not mistaken.

Reply:
that is less convenient api for scala, which I think it used more. We talked about this when we added resources into TaskContextImpl class for the original resource scheduling and went with this approach so am keeping it common with that.  I would prefer to keep consistent.

Reply:
Hm .. I am still hesitant about having different methods for Java and Scala in particular given that It is legitimate for Scala side to use Java side (the opposite is not).

My impression of Spark APIs are in general same or similar API usage across languages. There are many examples that uses Java classes, for example, in Structured Streaming as an example. e.g) `StateOperatorProgress`, `StreamingQueryProgress`, etc.

So, we should probably keep it consistent with other instances, instead of `TaskContextImpl`. Seems we marked `resourceJMap` as `Evolving` so looks not too late to change .. though ..

Reply:
But .. should be okay to keep it as is for now .. we can discuss and do this in a separate JIRA too.

Reply:
I guess I disagree with you and others already agreed on the previous PR for TaskContextImpl.  why not just have the function and its friendly to both java and scala api's. We have lots of java specific versions of apis. Go look at the DataSet api docs and look for "Java-specific". I agree that there is no set rule for this though, but I would also rather make it more friendly for the common case which I think is the scala api.  One point here is that I don't think Spark has many api that return things, instead its the input parameters.  there is collect() and collectAsList() but I think when I was looking for TaskContextImpl I didn't find any other usages

 @mengxr  @jiangxb1987  as they were in on original api.

Reply:
I was not so against it because I took it as just one or several exceptions. If this is going to be a pattern, I think it is a problem. Why not have one function legitimate for both Scala and Java that gives you a consistent API usage. Other "java specific" have the same names and most of then are just overloading. See `functions.scala` as an example.

I would like to run this separately though rather then setting up the policy here which might be slightly off topic.

Reply:
None of the java specific functions in functions.scala return a different type, they take different types. That is the difference. You can't overload a function based on return type in scala.

Reply:
My point is the most of java specifics are like that. I think I pointed out some examples that return Java instances in structured streaming. I have an idea to kind of compromising. I will send an email to the dev list tomorrow. We could discuss further there. For now, I am okay with keeping this line in this PR.